js_test/tasks.ts

```diff
@@ -0,0 +1,303 @@
+    let Message = await getExpectedMessage(Messages)
+    if (Message == null) {
+        for (let msg of Messages) {
+            console.log(msg)
+            console.log(msg.Tags)
+        }
+    }
+    if (getTag(Message, "Error")) {
+        throw getTag(Message, "Error")
+    }
+    return Message.Data
+}
+async function testReportAllResult(nodes: string[], taskId: string, signer: any) {
+    for (const node of nodes) {
+        let res = await testReportResult(node, taskId, signer)
+        console.log(`${node} result: ${res}`)
+    }
+}
+async function testBalance(address: string, signer: any) {
+    let action = "Balance"
+
+    let msgId = await message({
+        "process": TOKEN_PROCESS,
+        "signer": signer,
+        "tags": [
+            {"name": "Action", "value": action},
+            {"name": "Recipient", "value": address},
+        ]
+    });
+
+    let Result = await result({
+        "process": TOKEN_PROCESS,
+        "message": msgId,
+    });
+
+    if (Result.Error) {
+        console.log(Result.Error)
+    }
+    let Messages = Result.Messages
+    if (getTag(Messages[0], "Error")) {
+        throw getTag(Messages[0], "Error")
+    }
+    console.log("Balance: ", Messages[0].Data)
+    return Messages[0].Data
+}
+async function testAllowance(signer: any) {
+    let action = "Allowance"
+
+    let msgId = await message({
+        "process": TASK_PROCESS,
+        "signer": signer,
+        "tags": [
+            {"name": "Action", "value": action},
+        ]
+    });
+    let Result = await result({
+        "process": TASK_PROCESS,
+        "message": msgId,
+    });
+    if (Result.Error) {
+        console.log(Result.Error)
+    }
+    let Messages = Result.Messages
+    if (getTag(Messages[0], "Error")) {
+        throw getTag(Messages[0], "Error")
+    }
+    // console.log(Messages)
+    console.log("allowance: ", Messages[0].Data)
+    return Messages[0].Data
+}
+async function testWithdraw(quantity: string, signer: any) {
+    let action = "Withdraw"
+
+    let msgId = await message({
+        "process": TASK_PROCESS,
+        "signer": signer,
+        "tags": [
+            {"name": "Action", "value": action},
+            {"name": "Quantity", "value": quantity},
+        ]
+    });
+    let Result = await result({
+        "process": TASK_PROCESS,
+        "message": msgId,
+    })
+    if (Result.Error) {
+        console.log(Result.Error)
+    }
+    let Messages = Result.Messages
+    if (getTag(Messages[0], "Error")) {
+        throw getTag(Messages[0], "Error")
+    }
+    let Message = await getExpectedMessage(Messages)
+    console.log("withdraw: ", Message.Data)
+    return Message.Data
+}
+
+function sleep(ms: number) {
+    return new Promise((resolve) => setTimeout(resolve, ms))
+}
+
+async function main() {
+    const signer = await getSigner()
+    let address = await getWalletAddress()
+    const nodes = ["js_aos1", "js_aos2", "js_aos3"]
+
+    await testAddWhiteList(address, signer)
```

Comment:
Calling this directly will return `Forbitten to operate white list`, should `setNodeRegistryManager` be called first? @xiaojiqing